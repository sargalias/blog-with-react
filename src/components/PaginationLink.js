import React from 'react';

const PaginationLink = ({ url, text }) => (
  <li className="list-item">
    <a
      className="page-link"
      href={url}
    >
      { text }
    </a>
  </li>
);

export default PaginationLink;
