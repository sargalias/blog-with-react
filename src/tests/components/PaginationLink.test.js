import React from 'react';
import { shallow } from 'enzyme';
import PaginationLink from '../../components/PaginationLink';

test('PaginationLink should render correctly', () => {
  const wrapper = shallow(<PaginationLink url="testLink" text="testText" />);
  expect(wrapper).toMatchSnapshot();
});

test('PaginationLink should correctly use its url and text props', () => {
  const url = 'testUrl';
  const text = 'text text';
  const wrapper = shallow(<PaginationLink url={url} text={text} />);
  expect(wrapper.render().children('a').attr('href')).toBe(url);
  expect(wrapper.text()).toBe(text);
});
