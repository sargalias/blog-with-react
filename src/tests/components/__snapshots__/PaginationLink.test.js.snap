// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationLink should render correctly 1`] = `
<li
  className="list-item"
>
  <a
    className="page-link"
    href="testLink"
  >
    testText
  </a>
</li>
`;
